src/compute-client/src/protocol/history.rs

```diff
@@ -180,49 +179,92 @@ where
         live_dataflows.retain(|dataflow| dataflow.as_of != Some(Antichain::new()));
 
         // Record the volume of post-compaction commands.
-        let mut command_count = 1;
-        command_count += live_dataflows.len();
-        command_count += final_frontiers.len();
-        command_count += live_peeks.len();
-        command_count += live_cancels.len();
-        if !final_configuration.all_unset() {
-            command_count += 1;
-        }
-
-        self.metrics.reset();
+        // For commands that batch multiple entries, we count the number of entries instead of the
+        // number of commands. This is to ensure that history reduction is not invoked too often.
+        self.reduced_count = 0;
 
         // Reconstitute the commands as a compact history.
+        // When we update `self.metrics`, we need to be careful to not transiently report incorrect
+        // counts, as they would be ovservable by other threads.
+
```

Comment:
Above, you probably want to say "observable" instead of "ovservable". Not sure why GH did not show me the suggestion button, sigh.

Reply:
Probably because the you get the suggestion button only in the "Files changed" tab, not if you view the per-commit diff.